lib/slack/src/core.ts

```diff
@@ -0,0 +1,36 @@
+interface SlackReply {
+    response_type: String,
+    text: String
+}
+
+// Slack sends application/x-www-form-urlencoded data, example in exampleRequest.txt
+// Details at https://api.slack.com/interactivity/slash-commands#app_command_handling
+// This function expects this to be decoded in to an object (express does this
+// automatically very easily), example below.
+// {
+//   token: 'YOlSzbHSkwCPwdqZ3eHbHW1U',
+//   team_id: 'T02A80YKK',
+//   team_domain: 'redbadger',
+//   channel_id: 'D01R6L9NQ3S',
+//   channel_name: 'directmessage',
+//   user_id: 'U01QP0VJWJG',
+//   user_name: 'cedd.burge',
+//   command: '/get_manager',
+//   text: '@cedd.burge',
+//   api_app_id: 'A01SHH2QF8S',
+//   is_enterprise_install: 'false',
+//   response_url: 'https://hooks.slack.com/commands/T02A80YKK/2001166095842/CZprfzCQr6UoWSc5d0DiZivb',
+//   trigger_id: '2024969546880.2348032665.16dbba515458bbd713a6e1216cf16bbd'
+// }
+const getUserName = (body: any) => {
+    return body.text
+}
```

Comment:
If we were to actually write a type for `body`, I would expect that type to live out in the layer that deals with the slack api (maybe some kind of `infrastructure/slack.ts` module, if we're thinking ahead?).  I would not expect `core.ts` to be importing anything (not even types) from `infrastructure/`.

I propose moving this function into an `infrastructure/slack.ts` module. @StuartHarris might disagree though.

Reply:
(it's also worth noting that I've never used the onion architecture in anger, so I'm probably not the best reviewer for this)

Reply:
Yeah I had a think about this, as there is some very specific domain knowledge inside of core.js, which felt a bit weird to me as well. After some chin scratching, I decided that this was ok though. This entire api exists to integrate with slack, so to me it feels like 'core' should tightly couple to the slack things, being as that is the whole point of it. If there was a type for body, I don't think it would need to be referenced outside of core either.

This is just one train of thought that I had though, there are probably other ones ...